protobufs/livekit_agent.proto

```diff
@@ -21,154 +21,174 @@ option ruby_package = "LiveKit::Proto";
 
 import "livekit_models.proto";
 
-message WorkerInfo {
-    string id = 1;
-    string namespace = 2;
-    string version = 3;
-    string name = 4;
-    JobType type = 5;
-    ParticipantPermission allowed_permissions = 6;
+service AgentService {
+  rpc StartAgentJob(StartAgentJobRequest) returns (Job);
+  rpc StopAgentJob(StopAgentJobRequest) returns (Job);
 }
 
-message AgentInfo {
-    string id = 1;
-    string name = 2;
-    string version = 3;
+message StartAgentJobRequest {
+  JobType type = 1;
+  string room_name = 2;
+  optional string participant_identity = 3;
+  string namespace = 4;
+  string metadata = 5;
+}
+
+message StopAgentJobRequest {
+  string id = 1;
 }
 
 message Job {
-    string id = 1;
-    JobType type = 2;
-    Room room = 3;
-    optional ParticipantInfo participant = 4;
-    string namespace = 5;
+  string id = 1;
+  JobType type = 2;
+  Room room = 3;
+  optional ParticipantInfo participant = 4;
+  string namespace = 5;
+  string metadata = 6;
+  JobStatus status = 7;
+  string error = 8;
+  float load = 9;
+  int64 started_at = 10;
+  int64 ended_at = 11;
+  int64 updated_at = 12; 
```

Comment:
i'm not sure we want any of these on `Job` - this is the job description passed to the worker to initialize a job.

Reply:
This mixes Job Model and Execution status in the same message indeed. This is the pattern that is also used by Egress. Most consumers of the Job object need its current or latest state as well, which makes it convenient to have a message that makes it easy to send them together.

The IngressInfo message attempts to separate the Model and State somehow by packaging all the state related entries into a an IngressState message that is in turns included inside the Ingress message. The IngressState entry is null when the Ingress is used purely as model (which only happens at initialization).

An laternative here would be to flip the Job and JobState relationship compared to ingress:

```
message JobState {
   JobStatus status = 1;
   string error = 2;
   float load = 3;
   int64 started_at = 4;
   int64 ended_at = 5;
   int64 updated_at = 6;
  Job job = 7; 
}
```

Reply:
this makes the most sense to me because the job config is an immutable property of the running process but for the sake of consistency we could compromise and copy ingress so they're at least separate objects